app/models/user.rb

```diff
@@ -286,7 +286,13 @@ def visible_assessments(assessment_type: nil, assessment_id: nil)
     if self.section_id
       assessments = Assessment.left_outer_joins(:section_due_dates)
```

Comment:
I think this can be simplified a bit to just make it a bit clearer what the conditions we're looking for are. 

From what I remember, we want to return all assessments where:

```
section_due_date.is_hidden = false  # The specific case
OR 
section_due_date.is_hidden = null AND assessment.is_hidden = false  # The default case
```

So unless I'm missing something, I think checking for both `false` and `nil` together on line 289 is doing too much.

Does something like this work?:

```ruby
assessments = Assessment.left_outer_joins(:section_due_dates)
                        .where('section_due_dates.section_id': self.section_id)
assessments = assessments.where('section_due_dates.is_hidden': false)
                         .or('section_due_dates.is_hidden': nil, 'assessments.is_hidden': false))
```

Reply:
Note to self: we changed up the logic of what is shown so the comment above no longer applies